tests/hawtio-test-suite/src/main/java/io/hawt/tests/features/config/TestConfiguration.java

```diff
@@ -47,23 +66,30 @@ public static boolean browserHeadless() {
     }
 
     public static AppDeployment getAppDeploymentMethod() {
+        if (deployment != null) {
+            return deployment;
+        }
         if (hasProperty(APP_URL)) {
-            return new URLDeployment(getRequiredProperty(APP_URL));
+            return deployment = new URLDeployment(getRequiredProperty(APP_URL));
```

Comment:
This notation looks a bit strange for Java. Maybe you can just split lines if you really want to assign an object to `deployment` before return. Same for the following lines.